lib/file.js

```diff
@@ -33,6 +34,8 @@ class File extends EventEmitter {
       this.done = true
       this.emit('done')
     }
+
+    this._serviceWorker = torrent.client.serviceWorker
```

Comment:
this should not be needed.  pass the data port to _serve

Reply:
this isn't for the _serve functionality, but for checking if the worker is registered and activated when running `getStreamURL` or `streamTo`

Reply:
how would it get this far unless there was a request from a sw?

Reply:
`getStreamURL` or `streamTo` is something like `renderTo` you can call it on any file at any point, and all it does is returns/sets the URL for the specific file, it could be that some idiot decides to run this without a service worker, or that he doesn't activate his service worker even tho its registered

Reply:
oh...     yeah...     i dunno.    I think that breaks your desire to support multiple pages.   if a user has to use this method to populate the src of the video, then the wt instance is already in the page requesting the file

It might be more portable to other implementations if you made a new fn to getSwRelativeUrl() so the client can augment it with their chosen scope.  then again i'm also fine with assuming they used '/' for the chosen scope for the default impl

Reply:
I completely forgot, you are right on this, it needs a path relative to the service worker registration scope, not the current scope, idk how I missed this